pkg/server/tls/dynamic_source.go

```diff
@@ -217,11 +218,16 @@ func (f *DynamicSource) Healthy() bool {
 
 func (f *DynamicSource) tryRegenerateCertificate(ctx context.Context, nextRenewCh chan<- time.Time) error {
 	return wait.PollUntilContextCancel(ctx, f.RetryInterval, true, func(ctx context.Context) (done bool, err error) {
+		f.log.Info("try to generate a certificate")
```

Comment:
Does this require high-level logging (Info)? I think a "debug" log level would be more appropriate.

Reply:
TBH, I asked myself the same question. The issue requester and other people seem to have been misled by the error message, so I thought getting better observability of the process of generating the certificate would avoid such issues.